go/ct/evm_fuzz_test.go

```diff
@@ -0,0 +1,274 @@
+		defer referenceResultState.Release()
+		if err != nil {
+			t.Fatalf("failed to run test case in reference VM: %v", err)
+		}
+
+		if testeeResultState.Status != referenceResultState.Status {
+			t.Fatal("invalid result, status does not match reference status:", errorReportString(state, testeeResultState, referenceResultState))
+		}
+
+		// if result is other than running, further checks may be misleading
+		if testeeResultState.Status != st.Running {
+			return
+		}
+
+		if testeeResultState.Gas != referenceResultState.Gas {
+			t.Fatal("invalid result, gas does not match reference gas:", errorReportString(state, testeeResultState, referenceResultState))
+		}
+
+		// Hack: lfvm does a pc transformation, but for code smaller than the required jump the pc will point to a different location
+		// - Geth will point to pc+offset, whenever it is an overflow
+		// - Lfvm will point min(pc+offset, len(code))
+		if testeeResultState.Pc == uint16(len(opCodes)) &&
+			testeeResultState.Pc != referenceResultState.Pc {
+			testeeResultState.Pc = referenceResultState.Pc
+		}
+
+		if !testeeResultState.Eq(referenceResultState) {
+			t.Fatal("invalid result, result state does not match reference state:", testeeResultState.Diff(referenceResultState), errorReportString(state, testeeResultState, referenceResultState))
+		}
+	})
+}
+
+func fuzzVm(testee ct.Evm, f *testing.F) {
+
+	rnd := rand.New(0)
+
+	prepareFuzzingSeeds(f, rnd)
+
+	f.Fuzz(func(t *testing.T, opCodes []byte, gas int64, revision byte, stackBytes []byte) {
+		state, err := corpusInputDataToCtState(opCodes, gas, revision, stackBytes)
+		if err != nil {
+			t.Skip(err)
+		}
+
+		result, _ := testee.StepN(state, 1)
+		result.Release()
+	})
+}
+
+const (
+	fuzzIdealStackSize     int = 7  // < max pops in a single instruction
+	fuzzMaximumCodeSegment int = 33 // < 1 instruction with 32 data bytes
+)
+
+// prepareFuzzingSeeds is a helper function to be used by similar fuzzing tests
+// the arguments passed to the f.Add function needs to match the arguments
+// passed to the f.Fuzz function in type, position and number
+func prepareFuzzingSeeds(f *testing.F, rnd *rand.Rand) {
+
+	// every possible revision
+	for revision := MinRevision; revision <= NewestSupportedRevision; revision++ {
+		// every possible opCode, even if invalid
+		for op := 0x00; op <= 0xFF; op++ {
+			// Some gas values: this is a hand made sampling of interesting values,
+			// the fuzzer will generate more interesting values around these, the initial
+			// list just sketches a region of interest around which the fuzzer will generate
+			// more values. I found no measurable difference about being more accurate.
+			for _, gas := range []int64{0, 1, 6, 10, 1000, math.MaxInt64} {
+
+				// generate a code segment with the operation followed by 6 random values
+				ops := make([]byte, fuzzMaximumCodeSegment)
+				rand.Read(ops[:])
+				ops[0] = byte(op)
+
+				// generate a stack: the stack contains a mixture of values
```

Comment:
would it help to make use of `gen.StackGenerator` ?

Reply:
No, This fuzzing tests intentionally leave the initial seed non-exhaustive. The interesting cases are in these tests found through the fuzzing itself and not because of the initial seed.